src/net.c

```diff
@@ -54,7 +54,7 @@ extern loc_cmd* selected_locale;
 extern HANDLE dialog_handle;
 extern BOOL is_x86_32;
 static DWORD error_code, fido_len = 0;
-static BOOL force_update_check = FALSE;
+volatile static BOOL force_update_check = FALSE;
```

Comment:
My understanding is that volatile is for variables that can be altered outside of the code for the application (e.g. from an interrupt routine that would not be part of Rufus). So I'm not sure I understand why it should be needed here. Even if it is to be changed from another thread, the compiler is smart enough not to need volatile and a quick search for "is volatile needed for multithreaded" will show that it is unnecessary for that purpose. Therefore I have to ask: Have you experienced a specific issue here that makes you believe that volatile is needed?